frontend/src/components/form/inputs/select.tsx

```diff
@@ -0,0 +1,120 @@
+                                                                                          value,
+                                                                                          onValueSelected,
+                                                                                          optionsContainer,
+                                                                                          children,
+                                                                                          noneLabel = 'None',
+                                                                                          disabled,
+                                                                                          className,
+                                                                                          ...props
+                                                                                        }) => {
+  const blurUtil = useBlur();
+
+  const selectRef = useRef<HTMLDivElement | null>(null);
+  const optionsRef = useRef<HTMLDivElement | null>(null);
+
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const [hasSelectedItem, setHasSelectedItem] = useState<boolean>(false);
+
+  useEffect(() => {
+    setHasSelectedItem(false)
+    blurUtil.addListener(blur)
+  }, [])
+
+  const getOptions = (): Array<Item> => {
+    const values = [...parentOptions];
+    if (!required) values.push({
+      value: -1,
+      label: noneLabel
+    })
+    return values;
+  }
+
+  const buttonLabel = useMemo(() => {
+    if (value === undefined || value === -1) {
+      if (hasSelectedItem) return noneLabel;
+      else return placeholder;
+    }
+    return getOptions().find(o => o.value === value)?.label ?? placeholder
+  }, [value, parentOptions, required, hasSelectedItem, placeholder])
+  const buttonId = useMemo(() => `button-${ placeholder.toLowerCase().replace(' ', '-') }`, [placeholder])
+
+  const parentClasses = ["select", className]
+  if (label) parentClasses.push("has-label")
+
+  return <div id="aplose-input" className={ parentClasses.join(' ') } ref={ selectRef } { ...props }>
+    { label && <div id="label" className={ required ? 'required' : '' }>{ label }{ required && '*' }</div> }
+
+    <div id="input"
+         aria-disabled={ disabled }
+         className={ isOpen ? 'open' : '' }>
+      <select required={ required }
+              className="hide-real-input"
+              value={ value }>
+        <option></option>
+        { getOptions().map(o => <option value={ o.value === -1 ? undefined : value }
+                                        key={ o.value }>{ o.label }</option>) }
+      </select>
+
+      <div id={ buttonId }
+           onClick={ () => !disabled && setIsOpen(!isOpen) }
+           className={ 'button' + (!value && !hasSelectedItem ? ' placeholder' : '') }>
+        <p>{ buttonLabel }</p>
+        { !isOpen && <IonIcon icon={ caretDown }/> }
+        { isOpen && <IonIcon icon={ caretUp }/> }
+      </div>
```

Comment:
Always for semantic reasons, it could be better to use a `button` tag for a button 🙂